tests/template_tests/filter_tests/test_widgets.py

```diff
@@ -0,0 +1,20 @@
+from django import forms
+from django.template import Context, loader
+from django.test import SimpleTestCase
+
+
+class SampleForm(forms.Form):
+    is_company = forms.CharField(widget=forms.CheckboxInput())
+    email = forms.EmailField(widget=forms.EmailInput())
+    password = forms.CharField(widget=forms.PasswordInput())
+
+
+class FormsTestCase(SimpleTestCase):
+    def test_filters(self):
+        template = loader.get_template("filters.html")
+        test_form = SampleForm()
+        c = Context({"form": test_form})
+        html = template.render(c.flatten())
+        self.assertIn('Checkbox', html)
+        self.assertIn('Email', html)
+        self.assertIn('Password', html)
```

Comment:
We can explicitly test this property (without involving templates), e.g. in `tests/forms_tests/widget_tests/test_widget.py`.

Reply:
Super, Thanks for the guidance 🙏